apm-agent-core/src/test/java/co/elastic/apm/agent/metrics/builtin/CGroupMetricsTest.java

```diff
@@ -33,10 +34,11 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
 
 class CGroupMetricsTest {
 
-    private final MetricRegistry metricRegistry = new MetricRegistry(mock(ReporterConfiguration.class));
+    private final MetricRegistry metricRegistry = new MetricRegistry(spy(CoreConfiguration.class), mock(ReporterConfiguration.class));
```

Comment:
Why can't you keep using `mock`?

Reply:
Because calling `getMetricSetLimit()` on a mock returns 0 while on a spy it returns the default value.